### DISCUSSION TITLE
一个纯路人关于预计耗时的真实经验分享（雾

### DISCUSSION THREAD

**Discussion by panjd123** [RUCICS/LinkLab-2024-Assignment#5] [General]
背景：某助教，没学过链接（当时由于时间不够没教），只看文档（草稿版本的文档，而不是目前的，目前的文档把所有验题人没读懂的地方都写明白了），Copilot

| 部分 | 耗时 |
| ----- | ----- | 
| 泛读文档 ，理解链接是啥东西，理解 FLE 是啥东西，理解 `fle.hpp` 文件，通过任务一 | 0.5h |
| 写完任务二的测试点一，但是读不懂文档，不知道 entry 怎么填，卡了半天 | 0.5h |
| 改进文档，这下读懂了，通过测试点一 | 0.5h |
| **审错题了**，以为代码传给我的结构体的 `data` 段也是没有填 0 的，写了个**巨复杂**的塞0，其他部分跟着移动的算法 | **3.5h** |
| 改进文档，这下读懂了，通过任务二 | 0.5h |
| 一次性写任务三 ~ 任务五，通过 | 1.5h |
| 任务六要改写之前的逻辑，不过我预留了改动的框架，所以改起来比较快，通过任务六的测试点二 | 0.5h |
| `.bss` 段是什么鬼东西，看不懂 | 0.5h |
| 改进文档，这下读懂了，特判 `.bss` 段，但是还是有错 | 0.5h |
|  修复 `.bss` 错写成 `.bbs` 的重大 bug，通过任务六 | 0.5h |
| 总耗时 | 9h |
| 如果没有审错题 | 6h |
| 如果不参与文档改进  | 5h |

于是文档里的预计耗时按照两倍填了 10-15 小时

**Comment by andylizf** [comment]
@panjd123 加油呀，学链接的大哥哥！
